### PR TITLE
Miscellaneous changes for DFP and FW objects

### PR DESCRIPTION
**Pull request by unasra**
Changes done in this PR :

- Corrected Grammatical mistakes in Terraform files
- Added Named List definition to all Access Code UTs and Terraform Example file
- Updated template files to change the subcategory of FW objects to "Threat Defense"
- Modified the Resource names of all objects to be singular
- Added Retry Mechanism to Access Code to resolve "Cannot delete bypass code assigned to policy" issue
-  Resolved TODO UTs for Access Codes , Named List , Network List 
- Added Named List Definition to Security Policy UTs , wherever required
- Changed Default Redirects UT to TODO in Security Policy due to the absence of Custom Redirects
- Modified all File Names , Unit Tests and Function definition to have Singular Names

### CODE REVIEW

File: internal/service/fw/api_security_policies_resource_test.go [hunk 1]
```diff
@@ -641,9 +606,11 @@ resource "bloxone_td_security_policy" "test_safe_search" {
 }
 
 func testAccSecurityPoliciesRules(name, rulesAction, rulesData string) string {
+	listName1 := acctest.RandomNameWithPrefix("named-list")
+	listName2 := acctest.RandomNameWithPrefix("named-list")
```
Comment (mathewab) [inline internal/service/fw/api_security_policies_resource_test.go:610]: The names should be parameterized. Having it here calls update on these supporting resources unnecessarily.

File: internal/service/fw/api_access_codes_resource.go [hunk 2]
```diff
@@ -150,15 +152,24 @@ func (r *AccessCodesResource) Delete(ctx context.Context, req resource.DeleteReq
...
 	}
 
-	httpRes, err := r.client.FWAPI.
-		AccessCodesAPI.
-		DeleteSingleAccessCodes(ctx, data.AccessKey.ValueString()).
-		Execute()
-	if err != nil {
-		if httpRes != nil && httpRes.StatusCode == http.StatusNotFound {
-			return
+	err := retry.RetryContext(ctx, InternalDomainListOperationTimeout, func() *retry.RetryError {
```
Comment (mathewab) [inline internal/service/fw/api_access_codes_resource.go:155]: I would keep a separate timeout constant

```suggestion
	err := retry.RetryContext(ctx, AccessCodeOperationTimeout, func() *retry.RetryError {
```

File: internal/service/fw/model_atcfw_access_code.go [hunk 3]
```diff
@@ -86,7 +86,6 @@ var AtcfwAccessCodeResourceSchemaAttributes = map[string]schema.Attribute{
 	},
 	"policy_ids": schema.ListAttribute{
 		ElementType:         types.Int64Type,
-		Optional:            true,
```
Comment (mathewab) [inline internal/service/fw/model_atcfw_access_code.go:89]: If it is allowed in the API, and not here, can you write a detailed comment why we are doing it?

File: internal/provider/provider.go [hunk 4]
```diff
@@ -130,7 +130,7 @@ func (p *BloxOneProvider) Resources(_ context.Context) []func() resource.Resourc
 
 		dfp.NewDfpResource,
 
-		fw.NewSecurityPoliciesResource,
+		fw.NewSecurityPolicyResource,
```
Comment (mathewab) [inline internal/provider/provider.go:133]: AcessCodes, NamedLists, NetworkLists, InternalDomainLists all should be singular

**Comment by mathewab** [comment]
please update description with all the changes listed out.

File: internal/service/fw/api_security_policies_resource_test.go (hunk 1 above)
Comment (unasra) [reply internal/service/fw/api_security_policies_resource_test.go]: Resolved

File: internal/service/fw/api_access_codes_resource.go (hunk 2 above)
Comment (unasra) [reply internal/service/fw/api_access_codes_resource.go]: Fixed

File: internal/service/fw/model_atcfw_access_code.go (hunk 3 above)
Comment (unasra) [reply internal/service/fw/model_atcfw_access_code.go:89]: written a comment

File: internal/provider/provider.go (hunk 4 above)
Comment (unasra) [reply internal/provider/provider.go]: Updated

File: internal/service/fw/api_security_policies_resource_test.go (hunk 1 above)
Comment (mathewab) [reply internal/service/fw/api_security_policies_resource_test.go]: The change you made only removes the variable names. They are still initialized within this function. 
They should be initialized outside the function and passed as parameters

File: internal/service/fw/api_access_codes_resource.go
Comment (mathewab) [inline internal/service/fw/api_access_codes_resource.go:1]: Can you please also rename the files.

File: internal/service/fw/api_access_codes_resource_test.go
Comment (mathewab) [inline internal/service/fw/api_access_codes_resource_test.go:1]: Can you please rename the test file as well, and all the test functions in here too? 

Applies to all the other resources that were renamed.

File: internal/service/fw/api_access_codes_resource.go
Comment (unasra) [reply internal/service/fw/api_access_codes_resource.go:1]: Done

File: internal/service/fw/api_access_codes_resource_test.go
Comment (unasra) [reply internal/service/fw/api_access_codes_resource_test.go:1]: Done

File: internal/service/fw/api_security_policies_resource_test.go (hunk 1 above)
Comment (unasra) [reply internal/service/fw/api_security_policies_resource_test.go]: Added them to the UT that calls that particular function.

**Review by mathewab — APPROVED** [review]

**Pull request merged**